executor/adapter.go

```diff
@@ -558,7 +558,19 @@ func (a *ExecStmt) handlePessimisticDML(ctx context.Context, e Executor) error {
 }
 
 // UpdateForUpdateTS updates the ForUpdateTS, if newForUpdateTS is 0, it obtain a new TS from PD.
-func UpdateForUpdateTS(seCtx sessionctx.Context, newForUpdateTS uint64) error {
+func UpdateForUpdateTS(seCtx sessionctx.Context, newForUpdateTS uint64, useFuture bool) error {
```

Comment:
Can we get the newForUpdateTS with `future` outside this function, then call this function?
Then this function can be simpler and the logic is more clear.

Reply:
done